handlers/utils/utils.go

```diff
@@ -45,9 +44,6 @@ func addLabel(config map[string]interface{}, newLabels map[string]string) {
 	for key, value := range newLabels {
```

Comment:
Everywhere you actually use this addLabel function, you create a map on the fly to pass in a single key value pair. Might as well change the signature to accept a key and value and save the map crreation